public/scripts/report.js

```diff
@@ -1,20 +1,22 @@
 (function() {
+
+  var searchLocationButton = document.getElementById('searchLocation');
```

Comment:
move this down to the line above where you add the event listener, for consistency yo